shopinvader_sale_order_report_without_price/services/sale.py

```diff
@@ -0,0 +1,41 @@
+from odoo.addons.base_rest import restapi
+from odoo.addons.base_rest.components.service import to_bool
+from odoo.addons.component.core import Component
+
+
+class SaleService(Component):
+    _inherit = "shopinvader.sale.service"
+    _usage = "sales"
+
+    @restapi.method(
+        routes=[(["/<int:_id>/download"], "GET")],
+        input_param=restapi.CerberusValidator("_get_download_schema"),
+        output_param=restapi.BinaryData(required=True),
+    )
+    def download(self, _id, **params):
+        # Add the input_param schema to the download route
+        return super().download(_id, **params)
+
+    def _get_download_schema(self):
```

Comment:
this will be a problem if we want to add more validation to input_param. Normally to be clean we should call super()._get_download_schema(), but it doesn't exist in base module, so this works for now

Reply:
referenced here https://github.com/OCA/rest-framework/issues/203